packages/opentelemetry/src/internal/tracer.ts

```diff
@@ -304,3 +304,12 @@ const unknownToAttributeValue = (value: unknown): OtelApi.AttributeValue => {
   }
   return Inspectable.toStringUnknown(value)
 }
+
+export const withActiveSpan = <A, E, R>(effect: Effect.Effect<A, E, R>): Effect.Effect<A, E, R> => {
```

Comment:
This should be wrapped with Effect.suspend

Reply:
Yeah it makes sense as right now it has a side effect right?

Reply:
Yep